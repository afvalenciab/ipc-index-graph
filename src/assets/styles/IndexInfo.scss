.index--info {
  padding: 0px 50px;
  display: flex;
  box-sizing: border-box;
}