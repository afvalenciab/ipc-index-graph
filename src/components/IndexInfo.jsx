import React from 'react';
import { connect } from 'react-redux';
import CarouselData from './CarouselData';
import CarouselItemData from './CarouselItemData';
import '../assets/styles/IndexInfo.scss';

const IndexInfo = (props) => {
  const {
    openingIpcValue,
    maximumIpcValue,
    minimumIpcValue,
    closingIpcValue,
  } = props;
  return (
    <section className='index--info'>
      <CarouselData>
        <CarouselItemData nameCard='Apertura' ipcValue={openingIpcValue} />
        <CarouselItemData nameCard='Máximo' ipcValue={maximumIpcValue} />
        <CarouselItemData nameCard='Mínimo' ipcValue={minimumIpcValue} />
        <CarouselItemData nameCard='Cierre' ipcValue={closingIpcValue} />
      </CarouselData>
    </section>
  );
};

const mapStateToProps = (state) => {
  return {
    openingIpcValue: state.openingIpcValue,
    maximumIpcValue: state.maximumIpcValue,
    minimumIpcValue: state.minimumIpcValue,
    closingIpcValue: state.closingIpcValue,
  };
};

export default connect(mapStateToProps, null)(IndexInfo);
