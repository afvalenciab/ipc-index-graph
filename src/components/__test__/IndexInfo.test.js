import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import IndexInfo from '../IndexInfo';


describe('IndexInfo testing', () => {
  test('Match Snapshot', () => {
    const indexInfo = create(
      <ProviderMock>
        <IndexInfo />
      </ProviderMock>
    );
    expect(indexInfo.toJSON()).toMatchSnapshot();
  });
});
