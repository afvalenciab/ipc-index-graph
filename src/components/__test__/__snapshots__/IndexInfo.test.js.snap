// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexInfo testing Match Snapshot 1`] = `
<section
  className="index--info"
>
  <div
    className="carousel--data"
  >
    <div
      className="carousel--item_data"
    >
      <h2>
        Apertura
      </h2>
      <div
        className="item--data"
      >
        <p
          className="price"
        >
          -
        </p>
      </div>
    </div>
    <div
      className="carousel--item_data"
    >
      <h2>
        Máximo
      </h2>
      <div
        className="item--data"
      >
        <p
          className="price"
        >
          -
        </p>
      </div>
    </div>
    <div
      className="carousel--item_data"
    >
      <h2>
        Mínimo
      </h2>
      <div
        className="item--data"
      >
        <p
          className="price"
        >
          -
        </p>
      </div>
    </div>
    <div
      className="carousel--item_data"
    >
      <h2>
        Cierre
      </h2>
      <div
        className="item--data"
      >
        <p
          className="price"
        >
          -
        </p>
      </div>
    </div>
  </div>
</section>
`;
